spec/models/external_work_spec.rb

```diff
@@ -30,13 +30,35 @@
     URLS = ["http://the--ivorytower.livejournal.com/153798.html"]
 
     URLS.each do |url|
-      let(:valid_url) {build(:external_work, url: url)}
```

Comment:
We don't really use `URLS`, so let's make it more concise. You can replace the entire `context "valid urls"` block with:
```rb
context "for valid URLs" do
  [200, 301, 302, 307, 308].each do |status|
    context "returning #{status}" do
      let(:external_work) { build(:external_work) }

      before do
        WebMock.stub_request(:any, external_work.url).to_return(status: status)
      end

      it "saves" do
        expect(external_work.save).to be_truthy
      end
    end
  end
end
```

Reply:
Done! That's a much cleaner way of handling it